pkg/activator/handler/handler.go

```diff
@@ -31,32 +33,53 @@ type ActivationHandler struct {
 	Activator activator.Activator
 	Logger    *zap.SugaredLogger
 	Transport http.RoundTripper
+	Reporter  activator.StatsReporter
 }
 
 func (a *ActivationHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	namespace := pkghttp.LastHeaderValue(r.Header, activator.RevisionHeaderNamespace)
 	name := pkghttp.LastHeaderValue(r.Header, activator.RevisionHeaderName)
-	config := pkghttp.LastHeaderValue(r.Header, activator.ConfigurationHeader)
 
-	endpoint, status, err := a.Activator.ActiveEndpoint(namespace, config, name)
-
-	if err != nil {
-		msg := fmt.Sprintf("Error getting active endpoint: %v", err)
+	start := time.Now()
+	capture := &statusCapture{
+		ResponseWriter: w,
+		statusCode:     http.StatusOK,
+	}
 
+	ar := a.Activator.ActiveEndpoint(namespace, name)
+	if ar.Error != nil {
+		msg := fmt.Sprintf("Error getting active endpoint: %v", ar.Error)
 		a.Logger.Errorf(msg)
-		http.Error(w, msg, int(status))
+		http.Error(w, msg, int(ar.Status))
 		return
 	}
 
 	target := &url.URL{
 		Scheme: "http",
-		Host:   fmt.Sprintf("%s:%d", endpoint.FQDN, endpoint.Port),
+		Host:   fmt.Sprintf("%s:%d", ar.Endpoint.FQDN, ar.Endpoint.Port),
 	}
-
 	proxy := httputil.NewSingleHostReverseProxy(target)
 	proxy.Transport = a.Transport
-
 	util.SetupHeaderPruning(proxy)
+	proxy.ServeHTTP(capture, r)
+
+	// Report the metrics
+	httpStatus := capture.statusCode
+	duration := time.Now().Sub(start)
+	if numRetries := w.Header().Get(activator.ResponseCountHTTPHeader); numRetries != "" {
```

Comment:
Nit: This is numTries, not numRetries.

Reply:
Will fix.

Reply:
Though, the header says it is number of retries :)